src/transformers/modeling_tf_pytorch_utils.py

```diff
@@ -319,7 +320,13 @@ def load_pytorch_state_dict_in_tf2_model(
                     continue
             raise AttributeError(f"{name} not found in PyTorch model")
 
-        array = apply_transpose(transpose, pt_state_dict[name], symbolic_weight.shape)
+        try:
+            array = apply_transpose(transpose, pt_state_dict[name], symbolic_weight.shape)
+        except tf.errors.InvalidArgumentError:
+            if not ignore_mismatched_sizes:
```

Comment:
Maybe we could instead check shapes directly here but since there are many transposes here I decided to go for the simpler try except. (The only reason why this might not be ok is because InvalidArgumentError might cover more than just invalid shapes)